src/NumericalAlgorithms/Interpolation/InterpolatorRegisterElement.hpp

```diff
@@ -67,21 +98,59 @@ struct RegisterElement {
 /// - Removes: nothing
 /// - Modifies: nothing
 ///
+/// When this struct is used as an action, the `apply` function will perform the
+/// registration with the interpolator. However, this struct also offers the
+/// static member functions `perform_registration` and `perform_deregistration`
+/// that are needed for either registering when an element is added to a core
+/// outside of initialization or deregistering when an element is being
+/// eliminated from a core. The use of separate functions is necessary to
+/// provide an interface usable outside of iterable actions, e.g. in specialized
+/// `pup` functions.
 struct RegisterElementWithInterpolator {
+ private:
+  template <typename ParallelComponent, typename RegisterOrDeregisterAction,
+            typename DbTagList, typename Metavariables, typename ArrayIndex>
+  static void register_or_deregister_impl(
+      const db::DataBox<DbTagList>& /*box*/,
+      Parallel::GlobalCache<Metavariables>& cache,
+      const ArrayIndex& /*array_index*/) noexcept {
+    auto& interpolator =
+        *Parallel::get_parallel_component<::intrp::Interpolator<Metavariables>>(
+             cache)
+             .ckLocalBranch();
+    Parallel::simple_action<RegisterOrDeregisterAction>(interpolator);
+  }
+
+ public:
+  template <typename ParallelComponent, typename DbTagList,
+            typename Metavariables, typename ArrayIndex>
+  static void perform_registration(const db::DataBox<DbTagList>& box,
+                                   Parallel::GlobalCache<Metavariables>& cache,
+                                   const ArrayIndex& array_index) noexcept {
```

Comment:
None of the new functions use the box or array index (except to pass them to functions that themselves don't use them).  Is there some interface being conformed to that requires these arguments be accepted, or can they be dropped?

Reply:
These functions need to conform to the same interface as is used for the observer registration and deregistration, so that they can all be called from a `tmpl::for_each` in the `Algorithm.hpp` pup function, so I think I need to keep the argument.
I'll add a code comment explaining the reason for the function interface.